backend/tarsy/hooks/hook_context.py

```
@@ -440,21 +489,31 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
             logger.error(f"Failed to trigger stage execution hooks: {e}")
+            if self.allow_exceptions:
+                raise
         
-        return False  # Don't suppress exceptions
+        return False  # Don't suppress exceptions from the context body
 
 
 @asynccontextmanager
-async def stage_execution_context(session_id: str, stage_execution: StageExecution) -> AsyncContextManager[StageExecutionHookContext]:
+async def stage_execution_context(
+    session_id: str,
```

Ruff (0.14.8)

Unused function argument: `session_id`

(ARG001)